pkg/agent/agentserver/backend_manager.go

```diff
@@ -0,0 +1,130 @@
+/*
```

Comment:
Could you please add unit tests for this file, to cover cases like adding same connection, removing first, last connection, etc...

Reply:
The test was in server_test.go. Moved it to backend_manager_test.go to be clear.

Reply:
Also added test for "adding same connection".